Src/Sarif.PatternMatcher/SearchSkimmer.cs

```diff
@@ -581,14 +579,23 @@ private static void MergeDictionary(IList<Dictionary<string, FlexMatch>> mergeFr
 
             rank = fingerprint.GetRank();
 
-            return new Dictionary<string, string>()
+            // Add all fingerprints with no sensitive information by default.
+            var fingerprints = new Dictionary<string, string>()
             {
                 { SecretHashSha256Current, fingerprint.GetSecretHash() },
                 { AssetFingerprintCurrent, fingerprint.GetAssetFingerprint() },
-                { SecretFingerprintCurrent, fingerprint.GetSecretFingerprint() },
-                { ValidationFingerprintCurrent, fingerprint.GetValidationFingerprint() },
                 { ValidationFingerprintHashSha256Current, fingerprint.GetValidationFingerprintHash() },
             };
+
+            // Add fingerprints that expose sensitive data in plaintext
```

Comment:
>fingerprints [](http://example.com/codeflow?start=19&length=12)

This code path will redact the validation-relevant JSON that exposes credential passwords, etc.